crates/sui-core/src/streamer.rs

```diff
@@ -32,16 +35,31 @@ where
     F: Filter<T> + Clone + Send + Sync + 'static + Clone,
 {
     pub fn spawn(buffer: usize) -> Self {
-        let (tx, rx) = mpsc::channel::<T>(buffer);
+        let gauge = if let Some(metrics) = mysten_metrics::get_metrics() {
+            metrics.channels.with_label_values(&["streamer"])
+        } else {
+            // We call init_metrics very early when starting a node. Therefore when this happens,
+            // it's probably in a test.
+            mysten_metrics::init_metrics(&Registry::default());
+            mysten_metrics::get_metrics()
+                .unwrap()
+                .channels
+                .with_label_values(&["streamer"])
+        };
+
+        let (tx, rx) = mysten_metrics::metered_channel::channel(buffer, &gauge);
+        let gague_clone = gauge.clone();
```

Comment:
this is the meat of this PR

Reply:
Is it possible to extract the logic to a `fn monitored_channel(buffer_size: usize, name: &'static str)` in `mysten-metrics`? It would be useful elsewhere too.

Reply:
yeah I want to clean this up eventually, adding a TODO